contracts/MerkleDrop.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "hardhat/console.sol";
+import "./IDrop.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract MerkleDrop is IDrop, Ownable {
+    using SafeERC20 for IERC20;
+
+    address public override token;
+    bytes32 public override merkleRoot;
+    bool public initialized;
+    uint256 public expireTimestamp;
+
+    mapping(address => bool) public claimed;
+
+    function init(
+        address _owner,
+        address _token,
+        bytes32 _merkleRoot,
+        uint256 _expireTimestamp
+    ) external {
+        require(!initialized, "Drop already initialized");
+        initialized = true;
+        token = _token;
+        merkleRoot = _merkleRoot;
+        expireTimestamp = _expireTimestamp;
+        _transferOwnership(_owner);
+    }
+
+    /**
+     * @dev Implementation to claim token.
+     * @param _recipient Address of user who claims the drop.
+     * @param _amount Amount of tokens allotcated based on voting power.
+     * @param _merkleProof Proof of merkle root generated from recipient address and value
+     *
+     * This implementation is the way to claim tokens, that were allocated to accounts. This means
+     * that the recipient and the value to claim data is converted to merkle root and given to this contract
+     * at the time of deployment.
+     *
+     * To claim caller should pass the recipient address, amount value and proof, after verifying the
+     * proof, recipient and its value were right, we transfer the tokens and mark it as claimed.
+     */
+    function claim(
+        address _recipient,
```

Comment:
Why not we use `msg.sender` instead of extra param?

For instance, I can claim for another user with my gas which sounds not good without their permission!